From 177d89b6269fa5a7b43744dce384a5469cb52a76 Mon Sep 17 00:00:00 2001
From: Ondrej Jirman <megi@xff.cz>
Date: Sun, 21 May 2023 01:26:59 +0200
Subject: [PATCH 3/4] Input: goodix: Try to keep regulator enable/disable
 balanced

Failure handling is still not optimal in this driver, but let's
try to improve it.

Signed-off-by: Ondrej Jirman <megi@xff.cz>
---
 drivers/input/touchscreen/goodix.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/input/touchscreen/goodix.c b/drivers/input/touchscreen/goodix.c
index 410d97d0ccb9..89369e880ed0 100644
--- a/drivers/input/touchscreen/goodix.c
+++ b/drivers/input/touchscreen/goodix.c
@@ -1515,6 +1515,7 @@ static int goodix_resume(struct device *dev)
 
 		error = regulator_enable(ts->vddio);
 		if (error) {
+			regulator_disable(ts->avdd28);
 			dev_err(dev, "Regulator vddio enable failed.\n");
 			return error;
 		}
-- 
2.43.0

